File: exercises/5_saga/practice/src/main/java/fulfillment/Order.java
package fulfillment;

public class Order {
    private String orderId;
    private String customerId;
    private String itemSku;
    private int    quantity;
    private double totalAmount;

    public Order() {}

    public Order(String orderId, String customerId, String itemSku, int quantity, double totalAmount) {
        this.orderId     = orderId;
        this.customerId  = customerId;
        this.itemSku     = itemSku;
        this.quantity    = quantity;
        this.totalAmount = totalAmount;
    }

    public String getOrderId()     { return orderId; }
    public String getCustomerId()  { return customerId; }
    public String getItemSku()     { return itemSku; }
    public int    getQuantity()    { return quantity; }
    public double getTotalAmount() { return totalAmount; }

    @Override public String toString() {
        return String.format("Order{id=%s, sku=%s, qty=%d, amount=%.2f}",
            orderId, itemSku, quantity, totalAmount);
    }
}

File: exercises/4_cost_optimization/practice/src/main/kotlin/fulfillment/Order.kt
package fulfillment

// Defaults + @JvmOverloads give this a real no-arg constructor, which Temporal's default
// Jackson-based DataConverter needs to deserialize activity/workflow arguments via field access.
data class Order @JvmOverloads constructor(
    val orderId: String = "",
    val customerId: String = "",
    val itemSku: String = "",
    val quantity: Int = 0,
    val totalAmount: Double = 0.0
)
